### DISCUSSION TITLE
warning on installation with QEinstall.sh

### DISCUSSION THREAD

**Discussion by diegorasero** [nguyen-group/QE-SSP#4] [General]
![Capture](https://github.com/nguyen-group/QE-SSP/assets/36854852/11024584-3f68-4629-ab9f-e05e435dd126)
Dear All,

I am using the book to start in the world of QE.
For QE installation use the QEinstall.sh script. Almost at the end I get several warnings about the PATH (I am not an expert on this subject). I send an image of the terminal where the warnings appear.
Despite these warnings, the final lines seem to indicate that the installation was successful.
Is it possible to fix this "problem"?
Can these warnings make me unable to reproduce the examples in the book later?
I appreciate any help you can give me.

Best regard.

**Comment by Abishek52254** [comment]
It just means that the program may not be able to find those installed part later when running the program...it is recommended to put your installed files on path but it may not be of much issue...if you can go to the address of those files and put them somewhere the program can access it easily like somewhere inside the program folder for example...that should do it.

**Comment by nguyen-group** [comment]
Hi, these warnings might happen due to installing Jupyter. That means that you have successfully installed Jupyter, but the executable files are stored on /home/drasero-usco/.local/bin, which is not on $PATH environment variable. So you can not call directly Jupyter by typing command jupyter-lab. In order to call directly jupyter-lab on the Terminal, you need to set the $PATH environment variable for /home/drasero-usco/.local/bin in .barsh file as follows:
export PATH="${PATH}:/home/drasero-usco/.local/bin"
You also can check Google to how to set environment variable.
[https://unix.stackexchange.com/questions/713131/a-pip-install-for-jupyter-notebook-puts-its-binaries-in-local-bin-and-then-i](https://unix.stackexchange.com/questions/713131/a-pip-install-for-jupyter-notebook-puts-its-binaries-in-local-bin-and-then-i)
[https://superuser.com/questions/1372793/the-script-is-installed-in-directory-which-is-not-path](https://superuser.com/questions/1372793/the-script-is-installed-in-directory-which-is-not-path)
etc.

**Comment by diegorasero** [comment]
Ok. Thanks for the directions.Best regards, Enviado desde mi Huawei de Claro.-------- Mensaje original --------De: "Quantum Materials @ Tohoku University" ***@***.***>Fecha: mar., 5 sep. 2023, 8:40 a. m.Para: nguyen-group/QE-SSP ***@***.***>CC: diegorasero ***@***.***>, Manual ***@***.***>Asunto: Re: [nguyen-group/QE-SSP] warning on installation with QEinstall.sh (Discussion #4)
Hi, these warnings might happen due to installing Jupyter. That means that you have successfully installed Jupyter, but the executable files are stored on /home/drasero-usco/.local/bin, which is not on $PATH environment variable. So you can not call directly Jupyter by typing command jupyter-lab. In order to call directly jupyter-lab on the Terminal, you need to set the $PATH environment variable for /home/drasero-usco/.local/bin in .barsh file as follows:
export PATH="${PATH}:/home/drasero-usco/.local/bin"
You also can check Google to how to set environment variable.
https://unix.stackexchange.com/questions/713131/a-pip-install-for-jupyter-notebook-puts-its-binaries-in-local-bin-and-then-i
https://superuser.com/questions/1372793/the-script-is-installed-in-directory-which-is-not-path
etc.

**Comment by artnugraha** [comment]
Dear all, 

Thank you for pointing out the warning message. This issue in the case of diegorasero user should not be a problem because the pip3 command was executed before the addition of the .bashrc variable. In the QEinstall.sh script, just after the pip3 command you can actually find the following two lines:
``` 
echo 'export PATH="$HOME/.local/bin:$PATH"' >> ~/.bashrc
source ~/.bashrc
```
which are essential for adding your local directory to the environment variables.

Anyway, now we have created another script to avoid the warning message by installing necessary Python libraries and Jupyter lab using the default repositories of Ubuntu/Debian, NOT by pip3.

Please take a look at the script "QEinstall2.sh" via this URL:
https://github.com/nguyen-group/QE-SSP/blob/master/QEinstall2.sh
After downloading this alternative installation script, you can test it by the following command from the terminal:
```
bash QEinstall2.sh
```

Best regards,
Nugraha

**Comment by diegorasero** [comment]
Thanks for your explanation.

El dom, 1 oct 2023 a las 8:59, Ahmad Ridwan Tresna Nugraha (<
***@***.***>) escribió:

> Dear all,
>
> Thank you for pointing out the warning message. This issue in the case of
> diegorasero user should not be a problem because the pip3 command was
> executed before the addition of the .bashrc variable. In the QEinstall.sh
> script, just after the pip3 command you can actually find the following two
> lines:
>
> echo 'export PATH="$HOME/.local/bin:$PATH"' >> ~/.bashrc
> source ~/.bashrc
>
> which are essential for adding your local directory to the environment
> variables.
>
> Anyway, now we have created another script to avoid the warning message by
> installing necessary Python libraries and Jupyter lab using the default
> repositories of Ubuntu/Debian, NOT by pip3.
>
> Please take a look at the script "QEinstall2.sh" via this URL:
> https://github.com/nguyen-group/QE-SSP/blob/master/QEinstall2.sh
> After downloading this alternative installation script, you can test it by
> the following command from the terminal:
>
> bash QEinstall2.sh
>
> Best regards,
> Nugraha
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nguyen-group/QE-SSP/discussions/4#discussioncomment-7157733>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIZFYRD6IRX5COUDQNENTFTX5FZNZANCNFSM6AAAAAA3465V6M>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Son Diego Alejandro Rasero Causil
